### PR TITLE
CSE Machine: Remove old components and make compact components the default

### LINKED ISSUE #2845: CSE Machine: Duplicate component definitions in components and compactComponents folders
The current source code for the CSE Machine have both a "compactComponents" folder and a main "components" folder, with many components having duplicate code.

The CSE Machine primarily uses the components located in the "compactComponents" folder, and the main components are barely used as the only way to see them is to toggle the experimental button at the top of the CSE Machine.

It would be best to remove these main components and refactor the compact components to be the new main components instead.

### PR DESCRIPTION
**Pull request by CZX123**
### Description
This PR refactors the components inside the CSE Machine, such that the compact components are now the default components. Fixes #2845

### Changes
- [X] Renamed all classes and variables which mention `compact`.
- [X] Removed all old components, most notably the `Grid` component.
- [X] Removed the experimental button toggle inside the `SideContentCseMachine` component.

### CODE REVIEW

**Comment by RichDom2185** [comment]
Is this PR ready @CZX123 ? I got the email for the review but the status is still draft.

**Comment by CZX123** [comment]
Yup, it's ready, was just waiting for the checks first.

File: src/features/cseMachine/components/ArrayNullUnit.tsx [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 
 /** this classes encapsulates a null value in Source pairs or arrays */
 export class ArrayNullUnit extends Visible {
-  arrayUnit: ReferenceType;
-  referencedBy: ReferenceType[];
+  arrayUnit: ReferenceType | ReferenceType;
+  referencedBy: (ReferenceType | ReferenceType)[];
 
-  constructor(referencedBy: ReferenceType[]) {
+  constructor(referencedBy: (ReferenceType | ReferenceType)[]) {
```
Comment (RichDom2185) [inline src/features/cseMachine/components/ArrayNullUnit.tsx:15]: `ReferenceType | ReferenceType` is just `ReferenceType`. Could you clean this up? Thanks!

File: src/features/cseMachine/CseMachineLayout.tsx [hunk 2]
```diff
@@ -149,10 +130,6 @@ export class Layout {
       v.reset();
     });
     Layout.values.clear();
-    Layout.compactValues.forEach((v, d) => {
-      v.reset();
-    });
-    Layout.compactValues.clear();
```
Comment (RichDom2185) [inline src/features/cseMachine/CseMachineLayout.tsx:155]: Do we not need the `.forEach` in the new `Layout.values` anymore?

**Review by RichDom2185 — COMMENTED** [review]
Are we getting rid of the old grid system with `ArrayLevel`, `ArrowLane`, `FrameLevel`, and `Grid` entirely?

File: src/features/cseMachine/CseMachineLayout.tsx (hunk 2 above)
Comment (CZX123) [reply src/features/cseMachine/CseMachineLayout.tsx:155]: Yes, the loop seems to be unnecessary, I'll remove it.

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**